renderers/ebitengine/ebitengine.go

```diff
@@ -0,0 +1,484 @@
+	"log/slog"
+	"math"
+	"strings"
+	"unsafe"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/text/v2"
+	"github.com/hajimehoshi/ebiten/v2/vector"
+	"github.com/totallygamerjet/clay"
+)
+
+var (
+	whiteImage *ebiten.Image
+)
+
+func init() {
+	img := ebiten.NewImage(3, 3)
+	img.Fill(color.White)
+	whiteImage = img.SubImage(image.Rect(1, 1, 2, 2)).(*ebiten.Image)
+}
+
+type RendererData struct {
+	Screen *ebiten.Image
+	Fonts  []text.Face
+}
+
+func MeasureText(txt clay.StringSlice, config *clay.TextElementConfig, userData unsafe.Pointer) clay.Dimensions {
+	fonts := *(*[]text.Face)(userData)
+
+	font := fonts[config.FontId]
+
+	width, height := text.Measure(txt.String(), font, font.Metrics().HLineGap)
+	return clay.Dimensions{
+		Width:  float32(width),
+		Height: float32(height),
+	}
+}
+
+func ClayRender(rendererData *RendererData, renderCommands clay.RenderCommandArray) {
+	screen := rendererData.Screen
+	fonts := rendererData.Fonts
+	for i := int32(0); i < renderCommands.Length; i++ {
+		renderCommand := clay.RenderCommandArray_Get(&renderCommands, i)
+		boundingBox := renderCommand.BoundingBox
+		switch renderCommand.CommandType {
+		case clay.RENDER_COMMAND_TYPE_RECTANGLE:
+			config := &renderCommand.RenderData.Rectangle
+			if config.CornerRadius.TopLeft > 0 {
+				RenderFillRoundedRect(screen, boundingBox, config.CornerRadius.TopLeft, config.BackgroundColor)
+			} else {
+				vector.DrawFilledRect(
+					screen,
+					boundingBox.X,
+					boundingBox.Y,
+					boundingBox.Width, boundingBox.Height,
+					color.RGBA{
+						uint8(config.BackgroundColor.R),
+						uint8(config.BackgroundColor.G),
+						uint8(config.BackgroundColor.B),
+						uint8(config.BackgroundColor.A),
+					}, true,
+				)
+			}
+		case clay.RENDER_COMMAND_TYPE_TEXT:
+			config := &renderCommand.RenderData.Text
+			cloned := strings.Clone(config.StringContents.String())
+			font := fonts[config.FontId]
+
+			opts := &text.DrawOptions{}
+			opts.ColorScale.Scale(
+				config.TextColor.R/255,
+				config.TextColor.G/255,
+				config.TextColor.B/255,
+				config.TextColor.A/255,
+			)
+			opts.GeoM.Translate(float64(boundingBox.X), float64(boundingBox.Y))
+			text.Draw(screen, cloned, font, opts)
+		case clay.RENDER_COMMAND_TYPE_SCISSOR_START:
+			screen = rendererData.Screen.SubImage(image.Rect(
+				int(boundingBox.X), int(boundingBox.Y),
+				int(boundingBox.X+boundingBox.Width),
+				int(boundingBox.Y+boundingBox.Height),
+			)).(*ebiten.Image)
+		case clay.RENDER_COMMAND_TYPE_SCISSOR_END:
+			screen = rendererData.Screen
+		case clay.RENDER_COMMAND_TYPE_IMAGE:
+			config := &renderCommand.RenderData.Image
+			img := (*ebiten.Image)(config.ImageData)
+			opts := &ebiten.DrawImageOptions{}
+			opts.GeoM.Translate(float64(boundingBox.X), float64(boundingBox.Y))
+			screen.DrawImage(img, opts)
+		case clay.RENDER_COMMAND_TYPE_BORDER:
+			panic("not implemented")
+			//Clay_BorderRenderData *config = &renderCommand->renderData.border;
```

Comment:
Remove all of this since it doesn't relate to ebitengine

Reply:
I'm still seeing the commented SDL code

Reply:
interesting, I didn't push